### ISSUE TITLE
Structural response test indeterministic

### ISSUE THREAD

**Issue by armingeiser** [KratosMultiphysics/Kratos#2686]
@philbucher found an undeterministic behaviour of two tests in the StructuralMechanicsApplication - response functions:

```
======================================================================
FAIL: test_execution (__main__.TestAdjointDisplacementResponseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "structural_response_function_test_factory.py", line 97, in test_execution
    self.assertAlmostEqual(self.gradient[nodeId][0], 2046491212.095884, 12)
AssertionError: -928807255.0609658 != 2046491212.095884 within 12 places

Stdout:

> Synchronize primal and adjoint modelpart for response: dummy

> Starting primal analysis for response: dummy
> Time needed for solving the primal analysis =  0.04 s
> Time needed for calculating the response value =  0.0 s

> Starting adjoint analysis for response: dummy
> Time needed for solving the adjoint analysis =  0.12 s

======================================================================
FAIL: test_execution (__main__.TestAdjointStressResponseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "structural_response_function_test_factory.py", line 109, in test_execution
    self.assertAlmostEqual(self.gradient[nodeId][0], -0.4522249114784208, 12)
AssertionError: -0.47761638505058157 != -0.4522249114784208 within 12 places

Stdout:

> Synchronize primal and adjoint modelpart for response: dummy

> Starting primal analysis for response: dummy
> Time needed for solving the primal analysis =  0.04 s
> Time needed for calculating the response value =  0.0 s

> Starting adjoint analysis for response: dummy
> Time needed for solving the adjoint analysis =  0.12 s

----------------------------------------------------------------------
```

**Comment by armingeiser** [comment]
The tests run at @MFusseder and my machines.

First question:

- Is HDF5 compiled for the nightly tests on the server?  @philbucher @msandre

**Comment by loumalouomega** [comment]
It is giving always the same or it changes?, did you tried just one thread?

**Comment by msandre** [comment]
I think HDF5 is not compiled for nightly tests.

**Comment by philbucher** [comment]
- happens with one and with more than one thread
- no, HDF is not compiled in nightly tests
- results are different every time

**Comment by loumalouomega** [comment]
And only if HDF5 is compiled?

**Comment by philbucher** [comment]
The test needs hdf5, so yes

**Comment by armingeiser** [comment]
Ok thanks!
Next steps for me and @MFusseder 
- [ ] Create a minimal example that runs without HDF5
- [ ] Test if the error happens again at @philbucher's machine

**Comment by armingeiser** [comment]
The idea to test the adjoint stuff even without hdf5 is as following:

Keep the existing tests, but add a second test function that removes the HDF5 processes from the parameters object and transfers the primal results to the adjoint model part via python.

Like this the tests will also work if HDF5 is not available. 
This is possible for static cases (which is all we have currently)

**Comment by msandre** [comment]
I think it is not bad to have some small tests which don't depend on hdf5. But even in this case, we should try to set up nightly builds which include hdf5. The reason is that there is an increasing number of applications and tests which depend on hdf5 and it is not always simple to replace the hdf5 for testing.

**Comment by msandre** [comment]
One more comment, I needed something similar for a test of the TL element (to transfer from a primal to a adjoint model part). This is something we may need in many places so it would be good to consolidate our code into one utility which takes care of this in our different tests. At the moment I have a c++ version but it is still quite primitive.

**Comment by philbucher** [comment]
+1 for @msandre 
I guess such a utility would be good to have in the core

**Comment by armingeiser** [comment]
Such a utility sounds interesting, although if HDF5 is added to the nightly tests, the modification of our tests seems not so time critical anymore.

**Comment by armingeiser** [comment]
Closed with #2703
